File: server/models/app_log.go
package models

// LogLevel represents severity of a log entry.
type LogLevel string

const (
	LogLevelError LogLevel = "error"
	LogLevelWarn  LogLevel = "warn"
	LogLevelInfo  LogLevel = "info"
)

// LogCategory represents the subsystem that produced the log.
type LogCategory string

const (
	LogCategoryVoice       LogCategory = "voice"
	LogCategoryVideo       LogCategory = "video"
	LogCategoryScreenShare LogCategory = "screen_share"
	LogCategoryWS          LogCategory = "ws"
	LogCategoryAuth        LogCategory = "auth"
	LogCategoryGeneral     LogCategory = "general"
	LogCategoryFeedback    LogCategory = "feedback"
)

// AppLog represents a structured log entry stored in SQLite.
type AppLog struct {
	ID        string      `json:"id"`
	Level     LogLevel    `json:"level"`
	Category  LogCategory `json:"category"`
	UserID    *string     `json:"user_id"`
	ServerID  *string     `json:"server_id"`
	Message   string      `json:"message"`
	Metadata  string      `json:"metadata"`
	CreatedAt string      `json:"created_at"`
}

// AppLogFilter defines query parameters for listing logs.
type AppLogFilter struct {
	Level    string `json:"level"`
	Category string `json:"category"`
	Search   string `json:"search"`
	Limit    int    `json:"limit"`
	Offset   int    `json:"offset"`
}

File: client/src/types/applog.ts
/**
 * Application log types — structured runtime logs (errors/warnings)
 * exposed to admins via the Logs panel. Separate from audit logs:
 * applog is for engineering visibility, audit is for moderation.
 */

export type AppLogLevel = "error" | "warn" | "info";
export type AppLogCategory = "voice" | "video" | "screen_share" | "ws" | "auth" | "general" | "feedback" | "livekit";

export type AppLog = {
  id: string;
  level: AppLogLevel;
  category: AppLogCategory;
  user_id: string | null;
  server_id: string | null;
  message: string;
  metadata: string;
  created_at: string;
  username: string | null;
  display_name: string | null;
};
